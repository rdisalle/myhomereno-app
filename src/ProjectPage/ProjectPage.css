.projectInfo_list {
    list-style-type: none;
    padding-left: 0px;
    padding-right: 0px;
    border: 2px #F2AB6D solid;
    text-align: left;
    padding-left: 10px;
}

.projectPage_status {
    text-align: center;
    font-size: 20px;
}

.ProjectCommands {
    margin-top: 10px;
}

.Project_Link {
    color: white;
    font-size: 16px;
    text-decoration: none;
    border: 1px solid white;
    padding-right: 5px;
    padding-left: 5px;
    background-color: #F2AB6D;
    height: 40px;
    width: 100%;
}

.Project_Link:hover {
    background-color: #587AA6;
}

.Project_Delete {
    color: white;
    font-size: 16px;
    margin-top: 10px;
    width: 200px;
    height: 30px;
    background-color: #587AA6;
    border: none;
    font-family: 'Cairo', sans-serif;
}

.projectPage_name {
    font-size: 25px;
    text-align: center;
}