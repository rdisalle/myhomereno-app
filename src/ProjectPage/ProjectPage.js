import React, { Component } from 'react';
import './ProjectPage.css';
import { Link } from 'react-router-dom';
import MyHomeRenoContext from '../MyHomeRenoContext';
import config from '../config';
import PropTypes from 'prop-types';
import format from 'date-fns/format';

class ProjectPage extends Component {
    static defaultProps = {
        name: '',
        summary: '',
        estimated_cost: '',
        room: '',
        details: '',
        total_time: '',
        type: '',
        status: '',
      };

    static contextType = MyHomeRenoContext;

    handleClickDelete = e => {
        e.preventDefault()
        const projectId = this.props.id;
    
        fetch(`${config.API_ENDPOINT}/api/projects/${projectId}`, {
            method: 'DELETE',
            headers: {
            'content-type': 'application/json'
            },
        })
            .then(res => {
            if (!res.ok)
                return res.json().then(e => Promise.reject(e));
            return;
            })
            .then(() => {
            this.context.deleteProject(projectId);
            this.props.delete();
            })
            .catch(error => {
            console.error({ error });
            });
    };

  render() {
      const { id, name, summary, estimated_cost, room, details, total_time, type, status, date_created } = this.props;

    return (
        <div className="ProjectPage">
            <div className="projectInfo">
                <ul className="projectInfo_list">
                    <li className="projectPage_name">
                        {name}
                    </li>
                    <li className="projectPage_status">
                        {status}
                    </li>
                    <li className="projectPage_room">
                        Room: {room}
                    </li>
                    <li className="projectPage_summary">
                        {summary}
                    </li>
                    <li className="projectPage_type">
                        {type}
                    </li>
                    <li className="projectPage_estimated_cost">
                        Cost: {estimated_cost}
                    </li>
                    <li className="projectPage_total_time">
                        Time: {total_time}
                    </li>
                    <li className="projectPage_details">
                        {details}
                    </li>
                    <li>
                    <div className="project_date">
                    <div className="project_date-created">
                        Created:
                        {' '}
                        <span className="Date">
                        {format(new Date(date_created), "LLLL do yyyy")}
                        </span>
                    </div>
                    </div>
                    </li>
                </ul>
            </div>
            <div className="ProjectCommands">
                <Link className="Project_Link" to={`/view-project-estimates/${id}`}>
                    View Estimates
                </Link>
                <Link className="Project_Link" to={`/add-estimate/${id}`}>
                    Add Estimate
                </Link>
                <Link className="Project_Link" to={`/edit-project/${id}`}>
                    Edit Project
                </Link>
            </div>
            <button
                    className="Project_Delete"
                    type='button'
                    onClick={this.handleClickDelete}
                >
                    Delete
            </button>
        </div>
    );   
  }
}

ProjectPage.propTypes = {
    id: PropTypes.number,
    onDeleteProject: PropTypes.func,
    name: PropTypes.string.isRequired,
    summary: PropTypes.string.isRequired,
    estimated_cost: PropTypes.string.isRequired,
    room: PropTypes.string.isRequired,
    details: PropTypes.string.isRequired,
    total_time: PropTypes.string.isRequired,
    status: PropTypes.string.isRequired,
    type: PropTypes.string.isRequired,
  };

export default ProjectPage;

