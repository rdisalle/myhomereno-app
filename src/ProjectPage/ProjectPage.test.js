import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ProjectPage from './ProjectPage';

describe(`ProjectPage component`, () => {
  const props = {
    id: 1,
    name: 'test-class-name',
  };

  it('renders a .ProjectPage by default', () => {
    const wrapper = shallow(<ProjectPage />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders the ProjectPage given props', () => {
    const wrapper = shallow(<ProjectPage {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});