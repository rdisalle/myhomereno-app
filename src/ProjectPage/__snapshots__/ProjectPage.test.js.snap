// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectPage component renders a .ProjectPage by default 1`] = `
<div
  className="ProjectPage"
>
  <div
    className="projectInfo"
  >
    <ul
      className="projectInfo_list"
    >
      <li
        className="projectPage_name"
      />
      <li
        className="projectPage_estimated_cost"
      />
      <li
        className="projectPage_total_time"
      />
      <li
        className="projectPage_room"
      />
      <li
        className="projectPage_details"
      />
      <li
        className="projectPage_type"
      />
      <li
        className="projectPage_status"
      />
      <li
        className="projectPage_summary"
      />
    </ul>
  </div>
  <div
    className="ProjectCommands"
  >
    <Link
      className="Project_Link"
      to="/view-project-estimates/undefined"
    >
      View Estimates
    </Link>
    <Link
      className="Project_Link"
      to="/add-estimate/undefined"
    >
      Add Estimate
    </Link>
    <Link
      className="Project_Link"
      to="/edit-project/undefined"
    >
      Edit Project
    </Link>
    <button
      className="Course_Delete"
      onClick={[Function]}
      type="button"
    >
      Delete
    </button>
  </div>
</div>
`;

exports[`ProjectPage component renders the ProjectPage given props 1`] = `
<div
  className="ProjectPage"
>
  <div
    className="projectInfo"
  >
    <ul
      className="projectInfo_list"
    >
      <li
        className="projectPage_name"
      >
        test-class-name
      </li>
      <li
        className="projectPage_estimated_cost"
      />
      <li
        className="projectPage_total_time"
      />
      <li
        className="projectPage_room"
      />
      <li
        className="projectPage_details"
      />
      <li
        className="projectPage_type"
      />
      <li
        className="projectPage_status"
      />
      <li
        className="projectPage_summary"
      />
    </ul>
  </div>
  <div
    className="ProjectCommands"
  >
    <Link
      className="Project_Link"
      to="/view-project-estimates/1"
    >
      View Estimates
    </Link>
    <Link
      className="Project_Link"
      to="/add-estimate/1"
    >
      Add Estimate
    </Link>
    <Link
      className="Project_Link"
      to="/edit-project/1"
    >
      Edit Project
    </Link>
    <button
      className="Course_Delete"
      onClick={[Function]}
      type="button"
    >
      Delete
    </button>
  </div>
</div>
`;
